Dessert.java

```diff
@@ -1,37 +1,20 @@
 
 class Dessert {
-  CheeseCake myCheeseCake = new CheeseCake();
-  Brownie myBrownie = new Brownie();
+  	protected String ID;
+    protected String Name;
+    protected double Price;
+    protected String Description;
   
-  static class CheeseCake{
-    String ID = "DE1";
-    String Name = "Cheese Cake";
-    double Price = 6.5;
-    String Description = "Cheese Cake from delecious Mascarpone.";
-    public void display(){
-        System.out.println("#1: "+Name+ " " +"\nPrice: "+Price+"$");
-      	System.out.println(Description);
-        System.out.println();
+  	public Dessert(String ID, String Name, double Price, String Description){
+      this.ID = ID;
+      this.Name = Name;
+      this.Price = Price;
+      this.Description = Description;
     }
-  }
-  static class Brownie{
-    String ID = "DE2";
-    String Name = "Brownie";
-    double Price = 5.5;
-    String Description = "Brownie from the best Lindt chocolate.";
-    public void display(){
-        System.out.println("#2: "+Name+ " " +"\nPrice: "+Price+"$");
-        System.out.println(Description);
-        System.out.println();
-    }
-  }
-  public double getPrice(String name){
-      if(name == "Cheese Cake"){
-      return myCheeseCake.Price;
-      }
-      if(name == "Brownie"){
-      return myBrownie.Price;
-      }
-      return 0;
+  
+  	public void displayNumber(){
+       System.out.println("#"+ID+": "+Name+ " " +"\nPrice: "+Price+"$");
+       System.out.println(Description);
+       System.out.println();
```

Comment:
Ez a rész most nem megy igaz? Be tudod dobni ide a hibát?

Reply:
Lehet hogy package probléma lesz - sima class öröklésnél a protected az csak a saját package-ben levő más osztályok számára működik, ennek nézz utána jobban hogy megy az extends abstract keyword **nélkül**

Reply:
![image](https://github.com/Dendzsi/Java-Practice/assets/137180234/23a62b6a-f303-4b1e-abd0-4ec3e70eaad9)

Reply:
Ha így írom be akkor kapok egy "<identifier expected>" hibát

Reply:
Ahogy most van felötltve úgy nem dod hibát de biztos, hogy nem szép és lehet, hogy nem is működik 😄

Reply:
Oksa ennek utánanézek és leírom majd ha találok valami jót 🐐

Reply:
Oke, jo a mostani

Reply:
A fenti képen úgy tűnik a Colán belül történnek a dolgok, pont az lenne a lényeg, hogy a static classban semmi display-el kapcsolatos rész nincs, hanem átfolyik rajta - egy iktassunk be egy közös debug sessiont valamikor, hogy jobban utána tudjunk menni.

Reply:
![image](https://github.com/Dendzsi/Java-Practice/assets/137180234/be749869-b0cd-4bdb-b96f-d145bc908d70)